### PR TITLE
Parameter scatter plot

### PR DESCRIPTION
**Pull request by PaulJonasJost**
Parameter scatter plot for optimization.

### CODE REVIEW

**Review by dweindl — COMMENTED** [review]
Would be nice to include this plot in one of the example notebooks.

File: pypesto/visualize/parameters.py [hunk 1]
```diff
@@ -471,3 +474,91 @@ def parameters_correlation_matrix(
...
+    size: Tuple[float, float] = None,
+    show_bounds: bool = False,
+):
+    """
+    Plot scatter of optimized parameters.
+
+    Parameters
+    ----------
+    result:
+        Optimization result obtained by 'optimize.py'
```
Comment (plakrisenko) [inline pypesto/visualize/parameters.py:494]:
```suggestion
        Optimization result obtained by 'optimize.py'.
```

File: pypesto/visualize/parameters.py [hunk 2]
```diff
@@ -471,3 +474,91 @@ def parameters_correlation_matrix(
...
+    parameter_indices:
+        List of integers specifying the parameters to be considered.
+    start_indices:
+        List of integers specifying the multistarts to be plotted or
+        int specifying up to which start index should be plotted
```
Comment (plakrisenko) [inline pypesto/visualize/parameters.py:499]:
```suggestion
        int specifying up to which start index should be plotted.
```

File: pypesto/visualize/parameters.py [hunk 3]
```diff
@@ -471,3 +474,91 @@ def parameters_correlation_matrix(
...
+    ]
+    # compare start_indices with start_indices_finite and log a warning
+    if not np.all(start_indices == start_indices_finite):
+        logger.warning(
+            'Some start indices were removed due to inf values at the start.'
+        )
+    # put all parameters into a dataframe, where columns are parameters
+    parameters = [
+        result.optimize_result[i_start]['x'][parameter_indices]
+        for i_start in start_indices
```
Comment (plakrisenko) [inline pypesto/visualize/parameters.py:538]:
```suggestion
        for i_start in start_indices_finite
```

**Review by plakrisenko — APPROVED** [review]

File: pypesto/visualize/parameters.py [hunk 4]
```diff
@@ -471,3 +474,92 @@ def parameters_correlation_matrix(
...
+    result: Result,
+    parameter_indices: Union[str, Sequence[int]] = 'free_only',
+    start_indices: Optional[Union[int, Iterable[int]]] = None,
+    diag_kind: str = "kde",
+    suptitle: str = None,
+    size: Tuple[float, float] = None,
+    show_bounds: bool = False,
+):
+    """
+    Plot scatter of optimized parameters.
```
Comment (yannikschaelte) [inline pypesto/visualize/parameters.py:489]: a better verbal description of what the plot will show would be good.

**Review by yannikschaelte — APPROVED** [review]

**Pull request merged**